### ISSUE TITLE
Multiline command substitution does not work in interactive mode

### ISSUE THREAD

**Issue by simonrouse9461** [fish-shell/fish-shell#5916]
fish version: 3.0.2
system: Darwin
terminal: xterm-256color

Multiline command substitution works fine in a script:
```fish
echo (
    echo 1
    echo 2
)
```
but it gives an error if I type it interactively in command line:
```fish
> echo (
fish: Unexpected end of string, expecting ')'
echo (
     ^
```
In contrast, typing `echo "` will get no problem. I expected them to work in the same way.

**Comment by zanchey** [comment]
As a workaround, you can hit Alt-Enter in the default bindings to insert a newline.

**Comment by krobelus** [comment]
Fixed in 3.1.0.
